### ISSUE TITLE
bug: spherical flux divergence does not telescope against spherical_area_weights

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#247]
## Summary

`SphericalAdvection2D`'s flux-form tendency does not integrate to zero against `spherical_area_weights`, even for a closed domain where every boundary flux vanishes. A shallow-water model built on it therefore loses mass at the 1e-3 level, where the Cartesian equivalent conserves it to machine precision.

The likely cause is a mismatch of conventions rather than a stencil error: the cell area the operator implicitly divides by is not the one `spherical_area_weights` returns, so the telescoping sum that makes a finite-volume scheme conservative does not close.

Found while building the spherical shallow-water model for jejjohnson/somax#73.

## Reproduction

```python
import jax.numpy as jnp, numpy as np
from finitevolx import SphericalGrid2D, SphericalAdvection2D, spherical_area_weights

g = SphericalGrid2D.from_interior(48, 24, (0., 360.), (-80., 80.), R=6.371e6)
adv = SphericalAdvection2D(grid=g)
area = spherical_area_weights(g)
interior = (slice(1, -1), slice(1, -1))
rng = np.random.RandomState(0)

def wrap(a):                      # periodic in longitude
    return a.at[:, 0].set(a[:, -2]).at[:, -1].set(a[:, 1])

h = wrap(jnp.asarray(1000.0 + 10 * rng.randn(g.Ny, g.Nx)))

# Purely zonal flow on a zonally periodic grid: nothing can leave.
u = wrap(jnp.asarray(rng.randn(g.Ny, g.Nx)))
d = adv(h, u, jnp.zeros_like(u))
print(float(jnp.sum(area[interior] * d[interior]))
      / float(jnp.sum(area[interior] * jnp.abs(d[interior]))))
```

## Expected Behavior

For a closed domain the area-weighted integral of the flux-form tendency is zero to machine precision: interior face fluxes appear twice with opposite sign and cancel. That is what makes the flux form conservative and is what the Cartesian `Advection2D` delivers.

## Actual Behavior

| configuration | `sum(A·dh) / sum(A·|dh|)` |
|---|---|
| zonal flow only, periodic in longitude | -5.0e-03 |
| meridional flow only, `v = 0` on polar faces | 1.3e-03 |

Both should be ~1e-16. The zonal case is the more diagnostic of the two: with longitude periodic and `v = 0` there is no boundary at all for mass to cross, so a non-zero result cannot be a boundary-condition effect.

Downstream, a spherical shallow-water model started from a balanced solid-body rotation loses 1.06e-3 of its mass over six hours. That figure is **independent of the time step** (identical at `dt` = 900, 600, 300 and 150 s) and only weakly dependent on resolution (1.28e-3 at `ny=16`, 0.93e-3 at `ny=48`), which rules out time-integration error and points at the spatial operator.

## Suggested Investigation

Compare the cell area implied by the operator's own flux normalisation against `spherical_area_weights` (`R² cos(lat_T) dlon dlat`). A finite-volume scheme conserves `sum(A_cell · h)` only for the specific `A_cell` it divides fluxes by, so if the zonal flux is normalised with a face-centred `cos(lat)` while the diagnostic uses a T-point one, the sum will not close. Either the operator or the weights function should be adjusted so the pair is consistent, and a test should pin the telescoping property directly.

## Environment

- `finitevolx` v0.0.41, float32, jax 0.10.1

## Relationships

- Related: jejjohnson/somax#73 (spherical models, which document this as a known limitation rather than asserting exact conservation)
